libr/debug/p/native/linux/linux_debug.c

```diff
@@ -517,3 +521,1439 @@ RList *linux_desc_list (int pid) {
+	if (!p) goto error;
+
+	memcpy (p, reg_buff, rbytes);
+
+	return p;
+
+	error:
+		eprintf ("linux_get_fp_regset: error\n");
+		free (reg_buff);
+		return NULL;
+}
+
+static siginfo_t *linux_get_siginfo(RDebug *dbg) {
+	int ret;
+	siginfo_t *siginfo;
+
+	siginfo = R_NEW0 (siginfo_t);
+	if (!siginfo) return NULL;
+
+	ret = ptrace (PTRACE_GETSIGINFO, dbg->pid, 0, siginfo);
+
+	if (ret == -1 || !siginfo->si_signo) {
+		free(siginfo);
+		return NULL;
+	}
+
+	return siginfo;
+}
+
+static void get_map_address_space(char *pstr, ut64 *start_addr, ut64 *end_addr) {
+	char *pp;
+
+	pp = pstr;
+
+	*start_addr = strtoul (pp, &pp, 16);
+	pp++;   /*Skip '-' */
+	*end_addr = strtoul (pp, &pp, 16);
+}
+
+static void get_map_perms(char *pstr, ut8 *fl_perms) {
+	char *p_pstr;
+	ut8 flags;
+
+	flags = 0;
+	for (p_pstr = pstr ; *p_pstr ; p_pstr++) {
+		switch (*p_pstr) {
+		case 'r':
+			flags |= R_MEM;
+			break;
+		case 'w':
+			flags |= W_MEM;
+			break;
+		case 'x':
+			flags |= X_MEM;
+			break;
+		case 'p':
+			flags |= P_MEM;
+			break;
+		case 's':
+			flags |= S_MEM;
+			break;
+		case '-':
+			break;
+		}
+	}
+	
+	*fl_perms = flags;
+	
+	if (((flags & P_MEM) && (flags & S_MEM)) || (!(flags & R_MEM) && !(flags & W_MEM))) {
+		eprintf ("setting WRG_PERM\n");
+		*fl_perms = WRG_PERM;
+	}
+}
+
+static void get_map_offset(char *pstr, ut64 *offset) {
+	char *pp;
+	
+	pp = pstr;
+	*offset = strtoul (pp, &pp, 16);
+}
+
+
+static void get_map_name(char *pstr, char **name) {
+	*name = strdup (pstr);
+}
+
+static bool has_map_deleted_part(char *name) {
+	char deleted_str[] = "(deleted)";
+	int len_suffx;
+	int len_name;
+	int ret;
+	
+	len_name = strlen (name);
+	len_suffx = strlen (deleted_str);
+	
+	ret = strncmp (name + len_name - len_suffx, deleted_str, len_suffx);
+	
+	return ret ? false : true;
+}
+	
+
+static bool get_anonymous_value(char *keyw) {
+	keyw = strchr (keyw, ' ');
+
+	while (isspace (*keyw))
+		keyw++;
+
+	return *keyw != '0';
+}
+
+static bool has_map_anonymous_content(char *buff_smaps, ut64 start_addr, ut64 end_addr) {
+	char identity[80];
+	char *str;
+	char *p;
+	char *pp;
+	char *extern_tok;
+	char *keyw;
+	bool is_anonymous;
+
+	snprintf (identity, sizeof (identity), "%08"PFMT64x"-%08"PFMT64x"", start_addr, end_addr);
+	str = strdup (buff_smaps);
+
+	for (p = strtok_r (str, "\n", &extern_tok) ; p ; p = strtok_r (NULL, "\n", &extern_tok)) {
+		if (strstr (p, identity) != NULL) { 
+			for ( pp = strtok_r (NULL, "\n", &extern_tok); pp ; pp = strtok_r (NULL, "\n", &extern_tok)) {
+				if ((keyw = strstr (pp, "Anonymous:")) != NULL || (keyw = strstr (pp, "AnonHugePages:")) != NULL) {
+					is_anonymous = get_anonymous_value (keyw);
+					free (str);
+					return is_anonymous;
+				}
+			}
+		}
+	}
+	
+	free (str);
+	return 0;
+}
+
+static bool is_a_kernel_mapping(char *map_name) {
+	bool ret;
+
+	if (strcmp (map_name, "[vsyscall]") == 0 ||
+		strcmp (map_name, "[vvar]") == 0 ||
+		strcmp (map_name, "[vdso]") == 0) {
+		ret = true;
```

Comment:
No need for a var here. Just return true and drop the else block and return false there.